DuckDuckGo/Tab.swift

```diff
@@ -19,15 +19,35 @@
 
 import Core
 
+protocol TabObserver: class {
+ 
+    func didChange(tab: Tab)
+    
+}
+
 public class Tab: NSObject, NSCoding {
 
+    struct WeaklyHeldTabObserver {
+        weak var observer: TabObserver?
+    }
+    
     struct NSCodingKeys {
         static let link = "link"
         static let viewed = "viewed"
     }
 
-    var link: Link?
-    var viewed: Bool = true
+    var observersHolder = [WeaklyHeldTabObserver]()
```

Comment:
Can we make this private? I know it is used in a unit test but it feels like it should be hidden.